certora/specs/ghoVariableDebtToken-rayMulDiv-summarization.spec

```diff
@@ -0,0 +1,276 @@
+    require getUserCurrentIndex(onBehalfOf) <= index;
+
+    uint256 amountScaled = rayDiv_gst(amount,index);
+    uint256 prev_bal = scaledBalanceOf(e, onBehalfOf);
+
+    mint(e,user,onBehalfOf,amount,index);
+
+    mathint after_bal = scaledBalanceOf(e, onBehalfOf);
+    assert (after_bal <= prev_bal + amountScaled);
+}
+
+/*================================================================================
+  When calling Calling updateDiscountDistribution, if discountScaled>0 then the 
+  balance of the sender must decrease.
+  (This catches mutant 5)
+  =================================================================================*/
+rule discount_takes_place_in_updateDiscountDistribution__sender() {
+    env e;
+    address sender; address recipient; 
+    uint256 senderDiscountTokenBalance; uint256 recipientDiscountTokenBalance; uint256 amount;
+
+    uint256 sender_scaledBal_prev = scaledBalanceOf(sender);
+    uint256 discountScaled = get_discount_scaled(e,sender);
+    updateDiscountDistribution(e, sender, recipient,
+                               senderDiscountTokenBalance, recipientDiscountTokenBalance,
+                               amount);
+    uint256 sender_scaledBal_after = scaledBalanceOf(sender);
+
+    assert (discountScaled > 0 => sender_scaledBal_after<sender_scaledBal_prev);
+}
+
+
+/*================================================================================
+  When calling Calling updateDiscountDistribution, if discountScaled>0 then the 
+  balance of the recipient must decrease.
+  (This catches mutant 13)
+  =================================================================================*/
+rule discount_takes_place_in_updateDiscountDistribution__recipient() {
+    env e;
+    address sender; address recipient; 
+    uint256 senderDiscountTokenBalance; uint256 recipientDiscountTokenBalance; uint256 amount;
+
+    uint256 recipient_scaledBal_prev = scaledBalanceOf(recipient);   
+    uint256 discountScaled = get_discount_scaled(e,recipient);
+    updateDiscountDistribution(e, sender, recipient,
+                               senderDiscountTokenBalance, recipientDiscountTokenBalance,
+                               amount);
+    uint256 recipient_scaledBal_after = scaledBalanceOf(recipient);
+
+    assert (discountScaled > 0 => recipient_scaledBal_after<recipient_scaledBal_prev);
+}
+
+
+
+/*================================================================================
+  When calling Calling updateDiscountDistribution, the argumet 'amount' has influence
+  on the discountPercent of the sender.
+  (This catches mutant 10)
+  =================================================================================*/
+rule in_updateDiscountDistribution_amount_affects_sender_discount() {
+    env e;
+    address sender; address recipient; 
+    uint256 senderDiscountTokenBalance; uint256 recipientDiscountTokenBalance; uint256 amount_1;
+    require (sender != recipient);
```

Comment:
since it's a satisfy rule, i dont see why it's necessary

Reply:
First, it catches a mutant.
Second I believe that such rules should be added automatically by the prover - it's a kind of sanity check that checks that the function doesn't ignore any of it's arguments.